Fabric.Identity.API/Services/IdentityProviderConfigurationService.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Fabric.Identity.API.Configuration;
+using Fabric.Identity.API.Models;
+using IdentityServer4.Quickstart.UI;
+using Microsoft.AspNetCore.Http;
+
+namespace Fabric.Identity.API.Services
+{
+    public class IdentityProviderConfigurationService : IIdentityProviderConfigurationService
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        public IdentityProviderConfigurationService(IHttpContextAccessor httpContextAccessor)
+        {
+            _httpContextAccessor = httpContextAccessor ?? throw new ArgumentNullException(nameof(httpContextAccessor));
+        }
+
+        public ICollection<ExternalProvider> GetConfiguredIdentityProviders()
+        {
+            var schemes = _httpContextAccessor.HttpContext.Authentication.GetAuthenticationSchemes();
+
+            var providers = schemes
+                .Where(x => x.DisplayName != null && !AccountOptions.WindowsAuthenticationSchemes.Contains(x.AuthenticationScheme))
```

Comment:
Should there be a filter on whitespace/empty DisplayName?

Reply:
In this case, I don't think that is necessary. We are only trying to filter out authentication schemes like Bearer which will have a null display name.